src/main/lua/exasol/adapter/databricks/http_client.lua

```diff
@@ -52,95 +35,142 @@ local function new_socket_factory(params)
----@private
+---Create a new custom TCP socket factory depending ono request arguments.
+---@param args RequestArgs request arguments
+---@return SocketFactory? socket_factory the custom socket factory or `nil` if the default socket factory should be used
+---@private used in unit tests
 function M._create_socket_factory(args)
     if is_unencrypted(args.url) then
+        -- No custom socket factory required for unencrypted requests.
         return nil
     end
     local verify_tls_certificate = args.verify_tls_certificate == nil or args.verify_tls_certificate
     if verify_tls_certificate then
+        -- TLS certificate should be verified. We can use the default socket factory.
         return nil
     else
+        -- TLS certificate should be ignored. We need a custom socket factory.
         return new_socket_factory(get_socket_params(false))
     end
 end
 
+---Note about ltn12: The socket.http module uses ltn12 sources and sinks to send request body data and read response body data.
+---See this page for details about ltn12: http://lua-users.org/wiki/FiltersSourcesAndSinks
+
+---An ltn12 sink for reading the response body.
+---@alias ResponseBodySink fun(string, string): integer
+---Callback for retrieving the collected response body as a string.
+---@alias ResponseBodyGetter fun(): string
+
+---Creates an ltn12 sink for reading the response body as a string.
+---Based on https://github.com/lunarmodules/luasocket/blob/master/src/ltn12.lua#L224
+---@return ResponseBodySink, ResponseBodyGetter
+---@private used in unit tests
+function M._table_sink()
+    local result = {}
+    local function sink(chunk, err)
+        if err then
+            local exa_error =
+                    ExaError:new("E-VSDAB-28", "Error while receiving HTTP response: {{error}}", {error = err})
+            log.error(tostring(exa_error))
+            return 0
+        end
+        if chunk then
+            table.insert(result, chunk)
+        end
+        return 1
+    end
+    local function result_getter()
+        return table.concat(result, "")
+    end
+    return sink, result_getter
+end
+
+---An ltn12 source for sending a request body.
 ---@alias BodySource fun(): string?
 
 ---Creates an ltn12 source for the given string data or `nil` if the data is nil.
 ---Based on https://github.com/lunarmodules/luasocket/blob/master/src/ltn12.lua#L118
----See details about ltn12: http://lua-users.org/wiki/FiltersSourcesAndSinks
----@param data string? body data content
----@return BodySource? source data iterator for body content blocks
-local function create_source(data)
-    local BLOCKSIZE<const> = 2048
+---@param data string? optional body content
+---@param block_size integer? optional block size, defaults to 2048
+---@return BodySource? source the source or `nil` if the body is `nil`
+---@private used in unit tests
+function M._create_source(data, block_size)
+    local DEFAULT_BLOCKSIZE<const> = 2048
+    block_size = block_size or DEFAULT_BLOCKSIZE
```

Comment:
Maybe add a sanity check for parameter block_size. 
A block_size = 0 could cause some difficult errors...

Reply:
Good point! Done.